zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/KerberosInterpreter.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter;
+
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.zeppelin.annotation.ZeppelinApi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Interpreter wrapper for Kerberos initialization
+ *
+ * TODO: coming soon
+ * Please see,
+ * https://zeppelin.apache.org/docs/latest/development/writingzeppelininterpreter.html
+ *
+ * runKerberosLogin() method you need to implement that determine Zeppelin's behavior.
+ * startKerberosLoginThread() needs to be called inside the open() and
+ * shutdownExecutorService() inside close().
+ */
+public abstract class KerberosInterpreter extends Interpreter {
+
+  Integer kinitFailCount = 0;
+  protected ScheduledExecutorService scheduledExecutorService;
+  public static Logger logger = LoggerFactory.getLogger(KerberosInterpreter.class);
+
+  public KerberosInterpreter(Properties property) {
+    super(property);
+  }
+
+  @ZeppelinApi
+  protected abstract boolean runKerberosLogin();
+
+  public String getKerberosRefreshInterval() {
+    if (System.getenv("KERBEROS_REFRESH_INTERVAL") == null) {
+      return "1d";
```

Comment:
Yes so usually this is controlled by https://web.mit.edu/kerberos/krb5-1.12/doc/admin/conf_files/krb5_conf.html with the parameter called `renew_lifetime`. But in some cases, it may expire after Kerberos ticket lifetime (1 day always due to a JDK Bug https://bugs.openjdk.java.net/browse/JDK-8044500). Hence, if this "KERBEROS_REFRESH_INTERVAL" is not set by user I'm trying to return "1d".